feat(avm): alu fuzzer

IlyasRidhuan opened the pull request:
Adds a simple fuzzer for the ALU subtrace (witgen + check_relation + interactions). Also includes some machinery to add more fuzzers to the `harness/` subdirectory.

This tests the alu operations that interact via the `sel_exec_dispatch_alu` from execution (therefore excludes `truncate` for now).

### Steps to run
`cmake --build --preset fuzzing-avm --target harness_alu_fuzzer`
`cd build-fuzzing-avm`
`./bin/harness_alu_fuzzer`

### NOTE
This will report crashes until  #18192 is merged

dbanks12 committed:
feat(avm): alu fuzzer

IlyasRidhuan commented:
* **#18551** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18551?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18552** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18552?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18550** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18550?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18479** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18479?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18296** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18296?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18294** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18294?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18198** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18198?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18198?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `merge-train/avm`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

IlyasRidhuan commented on barretenberg/cpp/src/barretenberg/avm_fuzzer/.clangd:
```diff
@@ -0,0 +1,9 @@
+CompileFlags:
```
so lsp can use the fuzzing headers

defkit reviewed (approved):
:sunglasses:

MirandaWood reviewed (commented):
Just FYI to build you'll need a change in `barretenberg/cpp/src/barretenberg/avm_fuzzer/fuzz_lib/simulator.cpp` to bring down changes to `app_logic_return_values` in the merge train (I made an attempt [here](https://github.com/AztecProtocol/aztec-packages/pull/18570/files#diff-03949857c818013d99979cd92ab59df30efae827e3c8f608c4b562e83d6a0d49)!)

MirandaWood committed:
feat(avm): alu fuzzer input generation (#18570)

Following ##18479, randomly generates inputs and test cases for fuzzing
the ALU
